test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -1547,58 +1577,251 @@ defmodule Trento.SapSystemTest do
 
-    test "should deregister a single DB instance if no SR enabled" do
+    test "should deregister a single DB instance of two if no SR enabled, SAP system not registered" do
       sap_system_id = UUID.uuid4()
+      database_host_id = UUID.uuid4()
+      second_database_host_id = UUID.uuid4()
       deregistered_at = DateTime.utc_now()
+      db_instance_number_1 = "00"
+      db_instance_number_2 = "01"
 
-      instances =
-        [
-          %DatabaseInstanceRegistered{
-            instance_number: instance_number_1,
-            host_id: host_id
-          },
-          %DatabaseInstanceRegistered{
-            instance_number: instance_number_2
-          }
-        ] =
-        build_list(
-          2,
-          :database_instance_registered_event,
-          sap_system_id: sap_system_id,
-          system_replication: nil
-        )
+      db_sid = fake_sid()
 
       assert_events_and_state(
         [
           build(
             :database_registered_event,
-            sap_system_id: sap_system_id
+            sap_system_id: sap_system_id,
+            sid: db_sid
+          ),
+          build(
+            :database_instance_registered_event,
+            sap_system_id: sap_system_id,
+            host_id: database_host_id,
+            instance_number: db_instance_number_1,
+            system_replication: nil,
+            sid: db_sid
+          ),
+          build(
+            :database_instance_registered_event,
+            sap_system_id: sap_system_id,
+            host_id: second_database_host_id,
+            instance_number: db_instance_number_2,
+            system_replication: nil,
+            sid: db_sid
           )
-          | instances
         ],
         %DeregisterDatabaseInstance{
           sap_system_id: sap_system_id,
-          host_id: host_id,
-          instance_number: instance_number_1,
+          host_id: database_host_id,
+          instance_number: db_instance_number_1,
           deregistered_at: deregistered_at
         },
         %DatabaseInstanceDeregistered{
           sap_system_id: sap_system_id,
-          host_id: host_id,
-          instance_number: instance_number_1,
+          host_id: database_host_id,
+          instance_number: db_instance_number_1,
           deregistered_at: deregistered_at
         },
         fn sap_system ->
           assert %SapSystem{
                    database: %Database{
-                     instances: [%Instance{instance_number: ^instance_number_2}]
+                     sid: ^db_sid,
+                     deregistered_at: nil,
+                     instances: [%Instance{instance_number: ^db_instance_number_2}]
                    }
                  } = sap_system
         end
       )
     end
 
-    test "should deregister an ENQREP Application Instance" do
+    test "should deregister the primary instance, the entire database and then the secondary instance without deregistering the database a second time, SAP system not registered" do
```

Comment:
Are we sure we need to keep the "a second-time" wording? I think we will  soon forget about the mistake we made but this will stay in the codebase

Reply:
You're right, I will rephrase it